thunder/executors/nvfuserex_impl.py

```diff
@@ -2567,3 +2567,74 @@ def scaled_dot_product_flash_attention_grad(
     execution_transform=scaled_dot_product_flash_attention,
     grad_transform=scaled_dot_product_flash_attention_grad,
 )
+
+
+def _scaled_matmul_check(
+    a: TensorProxy,
+    b: TensorProxy,
+    scale_a: TensorLike,
+    scale_b: TensorLike,
+    bias: TensorLike | None = None,
+    scale_result: TensorLike | None = None,
+    out_dtype: dtypes.dtype | None = None,
+    use_fast_accum: bool = False,
+) -> bool:
+    from thunder.core.devices import to_torch_device
+
+    enable_matmul: None | bool = get_compile_option("nv_enable_matmul", "Enable nvFuser matmul.")
+    enable_linear: None | bool = get_compile_option("nv_enable_linear", "Enable nvFuser linear.")
+    if not (enable_matmul or enable_linear):
+        return False
+    if not (scale_a.numel == 1 and scale_b.numel == 1 and scale_result is None and not use_fast_accum):
+        return False
+    if not (
+        a.dtype in dtypes.float_8bit_dtypes
+        and b.dtype in dtypes.float_8bit_dtypes
+        and scale_a.dtype == dtypes.float32
+        and scale_b.dtype == dtypes.float32
+    ):
+        return False
+    if torch.cuda.get_device_capability(to_torch_device(a.device))[0] < 9:
+        return False
+    return True
+
+
+def _scaled_matmul(
+    a: TensorProxy,
+    b: TensorProxy,
+    scale_a: TensorLike,
+    scale_b: TensorLike,
```

Comment:
according to https://github.com/pytorch/ao/blob/200589beb900d4a39a880d4e31b942bed9d31f12/torchao/float8/float8_python_api.py#L22, these would be already inverse scaling factors.